JavaScript/Topics/3.Web_Browser/README.md

```diff
@@ -1 +1,176 @@
+
+The following code snippet is added inside the header content to link JavaScript with the HTML file:
+
+`<script src = "file_name.js"></script>`
+
+The js file is shown below:<br>
+``` JS
+
+"use strict";
+window.onload = function what() {
+    let x = document.getElementById("main");
+    x.innerHTML = "External JavaScript";
+};
+
+```
+
+The output is shown below, after running the HTML file.<br>
+![Output External JS](Code_Snippets/3.Output_External_Js.png)
+
+<br>
+
+<b>Note: </b><br>
+If there is no *window.onload* function in JS file, add the `<script>` tag at the end of the body content. Otherwise it will load before the body content and throw an error.
+
+```HTML
+
+<body>
+    <h1 id = "main"></h1>
+
+    <script src = "../JS/script.js"></script>
+</body>
+
+```
+Else, it can be added in the head content and JS can access the DOM elements if `window.onload` function is included.
+
+<br><hr><br>
+
+# How JavaScript files get compiled?
+
+In most low-level languages, the files get compiled and then executed accordingly. But in JavaScript, compilation takes place during the execution of the file. This type of compiler is known as Just-in-time(JIT) compiler.
+
+For example, V8 engine is used for compilation of JavaScript files. It is developed by Google.
+
+<br>
+
+## Working of JIT compiler
+
+The Source code of JS file is converted into the *byte code*, which is the intermediate conversion.<br> 
+This bytecode gets interpreted by Virtual Machine. Then it is converted to machine understandable code in section-wise (can be per function or per line) dynamically so that the program can run faster.
+
+## Advantages of JIT compiler
+
+1. Suited for dynamic programming languages.
+2. More secure
+3. The source code can be executed again without compilation. The files get cached and can be reused.
+4. Less memory usage.
+
+## Disadvantages of JIT compiler
+1. JIT reserve more cache memory.
+2. JIT requires more startup time when executed for the first time.
+
+<br><hr><br>
+
+# Why should JavaScript be chosen as the sole language for front-end web scripting? 
+
+1. JavaScript offers browser API as well as a graphical user interface, which HTML and CSS cannot accomplish.
+2. When implemented, JavaScript uses the notion of object-oriented programming to decrease the load on the server.
+3. JavaScript contains a large library of libraries, such as jQuery, AJAX, and others, that are useful for complex work/projects.
+4. Many front-end engineers prefer JavaScript because of its clear and straightforward syntax.
+5. JavaScript is rapidly improving as a result of its popularity, and its capacity to make web pages more interactive is growing by the day.
+   
```

Comment:
Also write javascript provides console Api, not in my way, after reading about it